src/reporter/index.js

```diff
@@ -100,10 +101,25 @@ export default class Reporter {
             // fixture, we can report this fixture start.
             nextReportItem = this.reportQueue[0];
 
-            await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
+            try {
+                await this.plugin.reportTestDone(reportItem.test.name, reportItem.testRunInfo, reportItem.test.meta);
+            }
+            catch (e) {
+                this.task.emit('error', new ReporterPluginError(e));
+            }
 
-            if (nextReportItem && nextReportItem.fixture !== currentFixture)
-                await this.plugin.reportFixtureStart(nextReportItem.fixture.name, nextReportItem.fixture.path, nextReportItem.fixture.meta);
+            if (nextReportItem && nextReportItem.fixture !== currentFixture) {
```

Comment:
[Quite nitpick and questionable] I think it can be easier to read if we use shortcuts:
```
if (!nextReportItem)
  continue;

if (nextReportItem.fixture === currentFixture)
  continue;
```

Reply:
I would prefer it if we extract the body of the loop in the separate method. In this case, we can replace `continue` with `return`.